src/lfortran/parser/fixedform_tokenizer.cpp

```diff
@@ -444,17 +444,21 @@ struct FixedFormRecursiveDescent {
         locations.push_back(loc);
     }
 
-    // Same as push_token(), but update `cur` and `t.cur`
+    // Same as push_token_no_advance(), but update `cur` and `t.cur`
     // TODO: just use `token_type`, and advance `cur` accordingly and check
     // that the string actually matches
-    // TODO: add push_token_no_advance() which does not modify cur
     void push_token2(unsigned char *&cur, const std::string &token_str,
```

Comment:
```suggestion
    void push_token_advance(unsigned char *&cur, const std::string &token_str,
```

I know you're doing this for the sake of not getting confused with overloading; for readability's sake I'd suggest using a precise meaning for the function(s):